config/routes.rb

```diff
@@ -668,9 +678,66 @@ def route_actions_hash
     actions: LEGACY_CRUD_ACTIONS - [:destroy] + [:show_past]
   )
 
+  namespace :herbaria do
+    resources :alls, only: [:index]
+    resources :curator_requests, only: [:new, :create]
+    resources :curators, only: [:create, :destroy], id: /\d+/
+    resources :merges, only: [:new]
+    resources :nexts, only: [:show], id: /\d+/
+    resources :nonpersonals, only: [:index]
+    resources :searches, only: [:index]
+  end
+  resources :herbaria, id: /\d+/
+  redirect_legacy_actions(
+    old_controller: "herbarium", new_controller: "herbaria",
+    actions: LEGACY_CRUD_ACTIONS - [:controller, :index, :show_past]
+  )
+
+  # Rails routes currently only accept template tokens
+  # rubocop:disable Style/FormatStringToken
+
+  # The immediately following "match" and "get" combine to redirect
+  # the legacy herbarium/delete_curator to the new herbaria/curators
+  # The "match" redirects
+  #   GET("/herbarium/delete_curator/nnn?user=uuu") and
+  #   POST("/herbarium/delete_curator/nnn?user=uuu")
+  # to
+  #   GET("/herbaria/curators/nnn?user=uuu")
+  # Therefore we need the following "get" to prevent
+  #   No route matches [GET] "/herbaria/curators/nnnnn"
+  match("/herbarium/delete_curator/:id",
+        to: redirect(path: "/herbaria/curators/%{id}"),
+        via: [:get, :post])
+  get("/herbaria/curators/:id", to: "herbaria/curators#destroy", id: /\d+/)
```

Comment:
I am concerned about the security implications of:
- the `match` because it routes both `get` and `post` to the same action; and
- the `get` because it writes to the db.

See https://guides.rubyonrails.org/v5.2/routing.html#http-verb-constraints

Reply:
Got it. Since i'm relatively out of the loop lately, let me ask: 

at this point how usable are the named path helpers like "herbaria_curators_path"... or are we still planning to move towards them? 

I believe with default routing the destroy action should work with a DELETE request to "herbaria_curator_path(:id)". But maybe the legacy requests are GETs, so you need to route it this way.

Reply:
I'd like to move toward the path helpers for the updated controllers.
I think the legacy requests are GETs. (I put a breakpoint at the top of Herbarium#destroy_herbarium, clicked `Destroy Fungarium` from an appropriate `show_herbarium` page, and -- when MO got to the breakpoint, called `request.method`:
```ruby
   141:   def destroy_herbarium
   142: byebug
...
(byebug) request.method
"GET"
```